### PR TITLE
Added multi-jumps to jump method/can_jump interval test

### PR DESCRIPTION
**Pull request by FriendlyGecko**
This is an attempt to start resolving #1848 .

As of current this only has added multi-jump capabilities to the jump method. I added in the ability to set various types of multi-jump: set, additive, and limited (additive to a point). 

I also added a jump interval timer delay based on ticks of the physics engine being updated.

### CODE REVIEW

**Comment by FriendlyGecko** [comment]
I believe that there is more work to be done on this. Specifically making it so players can jump off ladders. In the meantime, I would like some critiques on how it is currently coded so that we can ensure that this is inline with the rest of the library.

**Comment by pushfoo** [comment]
@FriendlyGecko are you still interested in working on this?

**Comment by FriendlyGecko** [comment]
Yes, I am.

However, I am unsure if it is ready/what do we need in order to determine if it is ready.

Items I am considering
- [x] Singular jump/multijump capable from one command
- [x] Multijump with configurable delay
- [x] Multiple styles of multijump
- [x] Capability to jump from ladder (without jumps instantly resetting)

Is there anything else you can think of that would be necessary?

**Comment by FriendlyGecko** [comment]
This request also adds an is_on_ground test which we can use to differentiate cases where we want altering behavior based on if the player is on a ladder vs the ground.

NOTE: In order to test this properly in regards to jumping while on a ladder, it is imperative that you alter the default jump_delay time to a number greater than 0 (10 is a good number). Otherwise player will launch like a rocket up the ladder with infinite jumps. The delay functions to limit the player from instantaneously jumping repeatedly and prevents them from just jumping up the entire ladder unchecked. 

To further support this, I believe that if we want to merge this, then we need to update the test AND change the default delay_time to 10.

File: arcade/physics_engines.py [hunk 1]
```diff
@@ -378,17 +381,17 @@ def is_on_ladder(self):
         # Check for touching a ladder
         if self.ladders:
             hit_list = check_for_collision_with_lists(self.player_sprite, self.ladders)
-            if len(hit_list) > 0:
+            if len(hit_list) > 0 and self.jump_ticks >= self.jump_delay:
```
Comment (pvcraven) [inline arcade/physics_engines.py]: A bit more 'pythonic' way of doing this, is just `if hitlist and`

File: arcade/physics_engines.py [hunk 2]
```diff
@@ -432,17 +450,51 @@ def disable_multi_jump(self):
...
+        to their air jump speed. """
+        if self.can_jump():
+            # Air Jump logic
+            if self.jumps_since_ground > 0:
+                # This checks if air_jump_velocity is set. If not it will default to the velocity for all air jumps.
+                if air_jump_velocity:
+                    air_jump = air_jump_velocity
+                else:
+                    air_jump = velocity
+                if air_jump_style == "additive":
```
Comment (pvcraven) [inline arcade/physics_engines.py:469]: Would it be better to use a constant/enum for this?>

**Review by pvcraven — CHANGES_REQUESTED** [review]
Any chance to add some unit tests for this? Documentation? Examples?

**Comment by FriendlyGecko** [comment]
> Any chance to add some unit tests for this? Documentation? Examples?

Perhaps not in near-near future, but early July my schedule should be a little more open.

By documentation do you mean within the code or within the rest of the project's documentation? And by examples do you mean like example code showing/explaining how one could use it? If so, then absolutely.

File: arcade/physics_engines.py (hunk 2 above)
Comment (FriendlyGecko) [reply arcade/physics_engines.py:469]: Looking into it.

Comment (FriendlyGecko) [reply arcade/physics_engines.py:469]: Does the current version of the code better fulfill your intent?

**Comment by einarf** [comment]
After some discussion: This feature can easily be implemented on the user side. I don't think we want more complexity in these simple physicsengines. We will probably brush them up a bit in 3.x. At that point we could look at something like this.

**Pull request closed without merging**